frontend/src/pages/DetailedCourse/DetailedCourse.tsx

```diff
@@ -78,15 +78,16 @@ const DetailedCourse: React.FC<IProps> = ({
       </MobileSearchWrapper>
       <InnerWrapper>
         <ImageWrapper imageUrl={commonCourseData.avatar} />
-        {isCourseApplicationSubmitted && status !== COURSE_STATUSES.pending && (
-          <ProgressBar
-            size="large"
-            text={progressText}
-            textColor={PROGRESS_COLOR}
-            variant={progressVariant}
-            value={progressValue}
-          />
-        )}
+        {(isCourseApplicationSubmitted && status !== COURSE_STATUSES.pending) ||
+          (status !== COURSE_STATUSES.rejected && (
```

Comment:
Let's move this condition before `return` statement. It will be better to understand the reason of comparison at first read